packages/javascript/bh-shared-ui/src/views/Explore/ContextMenu/AssetGroupMenuItemPrivilegeZonesEnabled.tsx

```diff
@@ -16,95 +16,153 @@
 
 import {
     Button,
+    Dialog,
     DialogActions,
     DialogContent,
     DialogDescription,
     DialogPortal,
     DialogTitle,
 } from '@bloodhoundenterprise/doodleui';
 import { MenuItem } from '@mui/material';
-import { FC } from 'react';
+import {
+    AssetGroupTagSelectorAutoCertifySeedsOnly,
+    AssetGroupTagTypeOwned,
+    HighestPrivilegePosition,
+    SeedTypeObjectId,
+} from 'js-client-library';
+import { FC, useState } from 'react';
+import { useMutation } from 'react-query';
 import { Link } from 'react-router-dom';
 
-const AssetGroupMenuItem: FC<{
-    assetGroupId: number;
-    assetGroupName: string;
-    disableAddNode: boolean;
-    isCurrentMember: boolean;
-    removeNodePath: string;
-    onAddNode?: (assetGroupId: string | number) => void;
-    onShowConfirmation?: () => void;
-    onCancelConfirmation?: () => void;
+import { isNode, useExploreSelectedItem, usePermissions, useTagsQuery, type NodeResponse } from '../../../hooks';
+import { useNotifications } from '../../../providers';
+import { detailsPath, labelsPath, privilegeZonesPath, zonesPath } from '../../../routes';
+import { Permission, apiClient } from '../../../utils';
+
+const ConfirmNodeChangesDialog: FC<{
+    dialogContent: string;
+    disableAccept: boolean;
+    onAccept: () => void;
+    onCancel: () => void;
+    open: boolean;
+}> = ({ dialogContent, disableAccept, onAccept, onCancel, open }) => {
+    return (
+        <Dialog open={open}>
+            <DialogPortal>
+                <DialogContent>
+                    <DialogTitle>Confirm Selection</DialogTitle>
+                    <DialogDescription>{dialogContent}</DialogDescription>
+                    <DialogActions>
+                        <Button variant='tertiary' onClick={onCancel}>
+                            Cancel
+                        </Button>
+                        <Button variant='primary' onClick={onAccept} disabled={disableAccept}>
+                            Ok
+                        </Button>
+                    </DialogActions>
+                </DialogContent>
+            </DialogPortal>
+        </Dialog>
+    );
+};
+
+export const AssetGroupMenuItem: FC<{
+    assetGroupType: 'tierZero' | 'owned';
```

Comment:
While this is the state of things today, PZ has taken a lot of pains to remove as much hardcode as possible with references to these names as technically the architecture by GA should support renaming of even this tags. With that in mind, what do you think of keeping the function handler prop for `onAddNode` and then pass that in while keeping this relatively how you have refactored it? 

That would keep it having that versatility while keeping props at a minimum?

Reply:
To add to my concern and give more context, there's already a concept of asset group tag type which is 'zone' or 'label' and this conflates that definition as well.